lib/Runtime/Library/WebAssembly.cpp

```diff
@@ -212,25 +212,26 @@ WebAssembly::ReadBufferSource(Var val, ScriptContext * ctx, _Out_ BYTE** buffer,
     const BOOL isTypedArray = Js::TypedArrayBase::Is(val);
     const BOOL isArrayBuffer = Js::ArrayBuffer::Is(val);
 
-    if (!isTypedArray && !isArrayBuffer)
-    {
-        *buffer = nullptr;
-        *byteLength = 0;
-        JavascriptError::ThrowTypeError(ctx, WASMERR_NeedBufferSource);
-    }
+    *buffer = nullptr;
+    *byteLength = 0;
 
     if (isTypedArray)
     {
         Js::TypedArrayBase* array = Js::TypedArrayBase::FromVar(val);
         *buffer = array->GetByteBuffer();
         *byteLength = array->GetByteLength();
     }
-    else
+    else if (isArrayBuffer)
     {
         Js::ArrayBuffer* arrayBuffer = Js::ArrayBuffer::FromVar(val);
         *buffer = arrayBuffer->GetBuffer();
         *byteLength = arrayBuffer->GetByteLength();
     }
+
+    if (*buffer == nullptr || *byteLength == 0)
```

Comment:
is it invalid for bytelength to be 0? probably unimportant, but i wonder if this case should be a WebAssemblyCompileError?